gradio/pipelines.py

```diff
@@ -18,253 +18,579 @@ def load_from_pipeline(pipeline: pipelines.base.Pipeline) -> dict:
-        }
-    elif hasattr(transformers, "VisualQuestionAnsweringPipeline") and isinstance(
-        pipeline, pipelines.visual_question_answering.VisualQuestionAnsweringPipeline
-    ):
-        pipeline_info = {
-            "inputs": [
-                components.Image(type="filepath", label="Input Image", render=False),
-                components.Textbox(label="Question", render=False),
-            ],
-            "outputs": components.Label(label="Score", render=False),
-            "preprocess": lambda img, q: {"image": img, "question": q},
-            "postprocess": lambda r: {i["answer"]: i["score"] for i in r},
-        }
-    elif hasattr(transformers, "ImageToTextPipeline") and isinstance(
-        pipeline,
-        pipelines.image_to_text.ImageToTextPipeline,  # type: ignore
-    ):
-        pipeline_info = {
-            "inputs": components.Image(
-                type="filepath", label="Input Image", render=False
-            ),
-            "outputs": components.Textbox(label="Text", render=False),
-            "preprocess": lambda i: {"images": i},
-            "postprocess": lambda r: r[0]["generated_text"],
-        }
-    elif hasattr(transformers, "ObjectDetectionPipeline") and isinstance(
-        pipeline, pipelines.object_detection.ObjectDetectionPipeline
-    ):
-        pipeline_info = {
-            "inputs": components.Image(
-                type="filepath", label="Input Image", render=False
-            ),
-            "outputs": components.AnnotatedImage(
-                label="Objects Detected", render=False
-            ),
-            "preprocess": lambda i: {"inputs": i},
-            "postprocess": lambda r, img: (
-                img,
-                [
-                    (
+                "outputs": components.AnnotatedImage(
+                    label="Objects Detected", render=False
+                ),
+                "preprocess": lambda i: {"inputs": i},
+                "postprocess": lambda r, img: (
+                    img,
+                    [
                         (
-                            i["box"]["xmin"],
-                            i["box"]["ymin"],
-                            i["box"]["xmax"],
-                            i["box"]["ymax"],
-                        ),
-                        i["label"],
-                    )
-                    for i in r
+                            (
+                                i["box"]["xmin"],
+                                i["box"]["ymin"],
+                                i["box"]["xmax"],
+                                i["box"]["ymax"],
+                            ),
+                            i["label"],
+                        )
+                        for i in r
+                    ],
+                ),
+            }
+        else:
+            raise ValueError(f"Unsupported pipeline type: {type(pipeline)}")
+
+    elif "diffusers.pipelines" in str(type(pipeline)):
```

Comment:
Ditto here – `_handle_diffusers_pipeline()`?, and ditto here with using `startswith` instead of substring?